exchanges/mock/recording_test.go

```diff
@@ -81,130 +84,122 @@ type TestStructLevel1 struct {
+	OkayVal2 float64        `json:"okayVal2"`
+	OkayVal3 map[string]any `json:"okayVal3"`
+	OkayVal4 map[string]any `json:"okayVal4"`
+	OkayVal5 []any          `json:"okayVal5"`
+	BadVal   int64          `json:"receiver_name"`
+	BadVal2  string         `json:"account_number"`
+	BadVal3  []any          `json:"secret"`
 }
 
-type TestStructLevel3 struct {
-	OkayVal  string  `json:"okayVal"`
-	OkayVal2 float64 `json:"okayVal2"`
-	BadVal   int64   `json:"receiver_name"`
-	BadVal2  string  `json:"account_number"`
+var testVal = []TestStructLevel0{
+	{
+		StringVal: "somestringstuff",
+		FloatVal:  3.14,
+		IntVal:    1337,
+		StructVal: TestStructLevel1{
+			OkayVal:  "stuff",
+			OkayVal2: 120938,
+			BadVal:   "CritcalBankingStuff",
+			BadVal2:  1337,
+			OtherData: TestStructLevel2{
+				OkayVal:  "stuff",
+				OkayVal2: 129219809899009009080980,
+				OkayVal3: map[string]any{"a": 1},
+				OkayVal4: map[string]any{},
+				OkayVal5: []any{},
+				BadVal:   1337,
+				BadVal2:  "Super Secret Password",
+				BadVal3:  []any{123, 'a'},
+			},
+		},
+		MixedSlice: []any{
+			[]map[string]any{{"id": 0}, {"id": 2}, {"id": 3}, {"id": 4}, {"id": 5}, {"id": 6}, {}},
+			[]any{float64(1586994000000), "6615.23000000", 'a', 1234, false, int64(17866372632), 0},
+			"abcd",
+		},
+	},
+	{
+		StringVal: "somestringstuff",
+		FloatVal:  3.14,
+	},
+	{
+		StringVal: "somestringstuff",
+		FloatVal:  3.14,
+		IntVal:    1337,
+	},
+	{
+		StringVal: "somestringstuff",
+		IntVal:    1337,
+	},
+	{},
+	{},
+	{},
 }
 
 func TestCheckJSON(t *testing.T) {
-	level3 := TestStructLevel3{
-		OkayVal:  "stuff",
-		OkayVal2: 129219,
-		BadVal:   1337,
-		BadVal2:  "Super Secret Password",
-	}
-
-	level2 := TestStructLevel2{
-		OkayVal:   "stuff",
-		OkayVal2:  129219,
-		BadVal:    0.222,
-		BadVal2:   1337888888,
-		OtherData: level3,
-	}
-
-	level1 := TestStructLevel1{
-		OkayVal:   "stuff",
-		OkayVal2:  120938,
-		BadVal:    "CritcalBankingStuff",
-		BadVal2:   1337,
-		OtherData: level2,
-	}
+	exclusionList, err := getExcludedItems()
+	require.NoError(t, err, "getExcludedItems must not error")
 
-	sliceOfPrimitives := []any{
-		[]any{float64(1586994000000), "6615.23000000"},
-		[]any{float64(1586994300000), "6624.74000000"},
-	}
+	data, err := json.Marshal(testVal)
+	require.NoError(t, err, "json.Marshal nust not error")
+	assert.NotNil(t, data, "json.Marshal nust not return nil")
 
-	testVal := TestStructLevel0{
-		StringVal:  "somestringstuff",
-		FloatVal:   3.14,
-		IntVal:     1337,
-		StructVal:  level1,
-		MixedSlice: sliceOfPrimitives,
-	}
+	var input any
+	err = json.Unmarshal(data, &input)
+	require.NoError(t, err, "json.Unmarshal must not error")
 
-	exclusionList, err := GetExcludedItems()
-	if err != nil {
-		t.Error("GetExcludedItems error", err)
-	}
-
-	vals, err := CheckJSON(testVal, &exclusionList)
-	if err != nil {
-		t.Error("Check JSON error", err)
-	}
+	vals, err := CheckJSON(input, &exclusionList, 4)
+	assert.NoErrorf(t, err, "Check JSON error: %v", err)
```

Comment:
Even though these are carry over messages from the prior code, can be adjusted to:

`assert.NoError(t, err, "CheckJSON should not error")`

For these and others